src/getsetall.jl

```diff
@@ -29,17 +31,60 @@ getall(obj, ::Properties) = getproperties(obj) |> values
 getall(obj, o::If) = o.modify_condition(obj) ? (obj,) : ()
 getall(obj, f) = (f(obj),)
 
+setall(obj, ::Properties, vs::Tuple) = setproperties(obj, NamedTuple{propertynames(obj)}(vs))
+setall(obj::NamedTuple{NS}, ::Elements, vs::Tuple) where {NS} = NamedTuple{NS}(vs)
+setall(obj::NTuple{N, Any}, ::Elements, vs) where {N} = (@assert length(vs) == N; Tuple(vs))
```

Comment:
Does this actually infer? How about annotating the return type with `::NTuple{N,Any}`?

Reply:
Or even better `::NTuple{N,eltype(vs)}`